server/controllers/explore_controller.js

```diff
@@ -80,3 +82,30 @@ export const populateExploreFeed = (user, conditions) => {
     return Articles.getArticlesFiltered(filter, conditions);
   });
 };
+
+export const updateUserArticleExploreData = (userId, articleId) => {
+  const promises = [];
+
+  return User.findById(userId).then((user) => {
+    const userVal = user.exploreNumber;
+    Article.findById(articleId).then((article) => {
+      const articleVal = article.avgUserScore;
+
+      if (userVal && userVal != 0) {
+        var promise1 = new Promise(function (resolve, reject) {
```

Comment:
I think you just want promise1 to equal the output of `Articles.updateArticleScore(article.id, userVal)`. The way you have it now, the promise never resolves since it never calls the `resolve` function.